src/tools/pkcs11-tool.c

```diff
@@ -2276,7 +2279,26 @@ static void sign_data(CK_SLOT_ID slot, CK_SESSION_HANDLE session,
 	fprintf(stderr, "Using signature algorithm %s\n", p11_mechanism_to_name(opt_mechanism));
 	memset(&mech, 0, sizeof(mech));
 	mech.mechanism = opt_mechanism;
-	hashlen = parse_pss_params(session, key, &mech, &pss_params);
+
+	if (opt_mechanism == CKM_RSA_PKCS_PSS) {
```

Comment:
You need these also for all the bunch of hashed PSS mechanisms otherwise the tests will fail:
```
=======================================================
SHA256-RSA-PKCS-PSS: Sign & Verify (KEY 01)
=======================================================
Using slot 0 with a present token (0x3818b2c5)
Using signature algorithm SHA256-RSA-PKCS-PSS
error: PKCS11 function C_SignInit failed: rv = CKR_ARGUMENTS_BAD (0x7)
Aborting.
====> ERROR: Failed to Sign data
```

Reply:
yes I'll revert that so it's like before. Didn't like parsing stuff for unrelated mechanisms.